src/Core/Content/Media/DataAbstractionLayer/MediaIndexer.php

```diff
@@ -97,16 +105,76 @@ public function handle(EntityIndexingMessage $message): void
             return;
         }
 
+        $context = $message->getContext();
+
+        $this->updateThumbnailsPath($context, $ids);
+
+        $this->updateThumbnailsRo($context, $ids);
+
+        $this->setMediaPaths($context, $ids);
+
+        $this->eventDispatcher->dispatch(new MediaIndexerEvent($ids, $context, $message->getSkip()));
+    }
+
+    public function getTotal(): int
+    {
+        return $this->iteratorFactory->createIterator($this->repository->getDefinition())->fetchCount();
+    }
+
+    public function getDecorated(): EntityIndexer
+    {
+        throw new DecorationPatternException(static::class);
+    }
+
+    private function updateThumbnailsPath(Context $context, array $ids): void
+    {
+        $mediaIdsWithMissingPaths = $this->connection->fetchFirstColumn(
+            'SELECT LOWER(HEX(id)) from media_thumbnail WHERE media_id IN (:ids) AND path IS NULL',
+            ['ids' => Uuid::fromHexToBytesList($ids)],
+            ['ids' => Connection::PARAM_STR_ARRAY]
+        );
+
+        if (\count($mediaIdsWithMissingPaths) === 0) {
+            return;
+        }
+
+        $query = new RetryableQuery(
+            $this->connection,
+            $this->connection->prepare('UPDATE `media_thumbnail` SET path = :path WHERE id = :id')
+        );
+
+        //get all media_thumbnails with missingPaths
         $criteria = new Criteria();
-        $criteria->addFilter(new EqualsAnyFilter('mediaId', $ids));
+        $criteria->addFilter(new EqualsAnyFilter('id', $mediaIdsWithMissingPaths));
+        $criteria->addAssociation('media');
 
-        $context = $message->getContext();
+        $all = $this->thumbnailRepository
+            ->search($criteria, $context)
+            ->getEntities();
+
+        /** @var MediaThumbnailEntity $mediaThumbnail */
+        foreach ($all as $mediaThumbnail) {
+            try {
+                $query->execute([
+                    'path' => $this->pathGenerator->generatePath($mediaThumbnail->getMedia(), $mediaThumbnail),
+                    'id' => Uuid::fromHexToBytes($mediaThumbnail->getId()),
+                ]);
+            } catch (\Throwable $e) {
+                // we need to catch errors thrown through getRelativeMediaUrl
+            }
```

Comment:
Please don't do that :'-(

Silently dropping exceptions can hide underlying issues. As a minimum the thrown exception should be logged somewhere. Additionally, I doubt it is necessary to catch ALL exceptions in the first place.